### PR TITLE
Adding kuttl tests

### PR DESCRIPTION
**Pull request by yuriolisa**
Signed-off-by: Yuri Sa <yurimsa@gmail.com>

### CODE REVIEW

File: config/samples/pets_v1_bestie.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: bestie
 spec:
   size: 3
-  image: quay.io/mkong/bestiev2
```
Comment (skattoju) [inline config/samples/pets_v1_bestie.yaml:7]: this is the operand which is not yet in opdev

**Review by skattoju — CHANGES_REQUESTED** [review]
the sample yaml should have the operand image not the operator image .. we should probably add ci to the operand repo as well lol https://github.com/makon57/bestie-l5-op

File: kuttl-test.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+kind: TestSuite
+crdDir: ./tests/_build/crds/
+artifactsDir: ./tests/_build/artifacts/
+kindContainers:
+  - local/bestie-operator:e2e
+commands:
+  - command: make generate
+  - command: make manifests
+  - command: make install
+  - command: kubectl get pods -n bestie
```
Comment (skattoju) [inline kuttl-test.yaml:11]: requires: 
make run & 
kubectl apply config/sample yaml

File: Makefile [hunk 3]
```diff
@@ -220,3 +227,117 @@ catalog-push: ## Push a catalog image.
...
+load-image-operator:
+	kind load docker-image local/bestie-operator:e2e
+
+.PHONY: start-kind
+start-kind:
+	kind create cluster --config $(KIND_CONFIG)
+
+.PHONY: create-crds
+create-crds:
+	kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagerconfigs.yaml ;\
```
Comment (skattoju) [inline Makefile:316]: shouldn’t we just install prometheus operator here ?

File: Makefile [hunk 4]
```diff
@@ -220,3 +227,117 @@ catalog-push: ## Push a catalog image.
...
+    kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_probes.yaml ;\
+    kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml ;\
+    kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_prometheusrules.yaml ;\
+    kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml ;\
+    kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml
+
+.PHONY: install-pgo
+install-pgo:
+	sudo git clone https://github.com/CrunchyData/postgres-operator-examples.git
+	kubectl apply -k postgres-operator-examples/kustomize/install/namespace
```
Comment (skattoju) [inline Makefile:328]: think we just need the default install ?

File: config/samples/pets_v1_bestie.yaml [hunk 5]
```diff
@@ -6,5 +6,5 @@ spec:
   size: 3
   image: quay.io/opdev/bestie
   maxReplicas: 10
-  version: "1.3"
-
+  replicas: 1
```
Comment (skattoju) [inline config/samples/pets_v1_bestie.yaml:9]: we don’t have replicas anymore

File: hack/install-pgo.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+sudo git clone https://github.com/CrunchyData/postgres-operator-examples.git
+
+kubectl apply -k postgres-operator-examples/kustomize/install/namespace
```
Comment (skattoju) [inline hack/install-pgo.sh:5]: do we need two different installs ? think just the default install ?

**Review by skattoju — APPROVED** [review]
looks good, mainly two comments: 1) should we install prometheus operator ? 2) need to remove replicas.

File: Makefile (hunk 4 above)
Comment (yuriolisa) [reply Makefile:328]: I thought also, however it didn't work if you don't perform the namespace.

File: hack/install-pgo.sh (hunk 6 above)
Comment (yuriolisa) [reply hack/install-pgo.sh:5]: No, I'll remove this.

**Review by skattoju — APPROVED** [review]
/lgtm

**Pull request merged**